File: app/src/main/java/com/capstone/project_niyakneyak/data/alarm_resource/AlarmRepository.java
package com.capstone.project_niyakneyak.data.alarm_resource;

import android.app.Application;

import androidx.lifecycle.LiveData;

import com.capstone.project_niyakneyak.data.alarm_model.Alarm;

import java.util.List;

public class AlarmRepository {
    private final AlarmDao alarmDao;
    private final LiveData<List<Alarm>> alarmsLiveData;

    public AlarmRepository(Application application) {
        AlarmDatabase db = AlarmDatabase.getInstance(application);
        alarmDao = db.alarmDao();
        alarmsLiveData = alarmDao.getAlarms();
    }

    public void insert(Alarm alarm) {
        AlarmDatabase.writeExecutor.execute(() -> {
            alarmDao.insert(alarm);
        });
    }

    public void update(Alarm alarm) {
        AlarmDatabase.writeExecutor.execute(() -> {
            alarmDao.update(alarm);
        });
    }

    public LiveData<List<Alarm>> getAlarmsLiveData() {
        return alarmsLiveData;
    }

    public void delete(int alarmId){
        AlarmDatabase.writeExecutor.execute(() -> {
            alarmDao.delete(alarmId);
        });
    }
}

File: app/src/main/java/com/capstone/project_niyakneyak/data/alarm_resource/AlarmRepository.kt
package com.capstone.project_niyakneyak.data.alarm_resource

import android.app.Application
import androidx.lifecycle.LiveData
import com.capstone.project_niyakneyak.data.alarm_model.Alarm

class AlarmRepository(application: Application) {
    private val alarmDao: AlarmDao?
    @JvmField
    val alarmsLiveData: LiveData<List<Alarm>>?

    init {
        val db: AlarmDatabase? = AlarmDatabase.getInstance(application)
        alarmDao = db?.alarmDao()
        alarmsLiveData = alarmDao?.alarms
    }

    fun insert(alarm: Alarm?) {
        AlarmDatabase.writeExecutor.execute { alarmDao!!.insert(alarm) }
    }

    fun update(alarm: Alarm?) {
        AlarmDatabase.writeExecutor.execute { alarmDao!!.update(alarm) }
    }

    fun delete(alarmId: Int) {
        AlarmDatabase.writeExecutor.execute { alarmDao!!.delete(alarmId) }
    }
}